rust/agama-lib/src/store.rs

```diff
@@ -21,20 +21,56 @@
 //! Load/store the settings from/to the D-Bus services.
 // TODO: quickly explain difference between FooSettings and FooStore, with an example
 
-use crate::base_http_client::BaseHTTPClient;
-use crate::bootloader::store::BootloaderStore;
-use crate::error::ServiceError;
-use crate::files::store::FilesStore;
-use crate::hostname::store::HostnameStore;
-use crate::install_settings::InstallSettings;
-use crate::manager::{InstallationPhase, ManagerHTTPClient};
-use crate::scripts::{ScriptsClient, ScriptsGroup};
-use crate::storage::http_client::ISCSIHTTPClient;
 use crate::{
-    localization::LocalizationStore, network::NetworkStore, product::ProductStore,
-    scripts::ScriptsStore, software::SoftwareStore, storage::StorageStore, users::UsersStore,
+    base_http_client::BaseHTTPClient,
+    bootloader::store::{BootloaderStore, BootloaderStoreError},
+    files::store::{FilesStore, FilesStoreError},
+    hostname::store::{HostnameStore, HostnameStoreError},
+    install_settings::InstallSettings,
+    localization::{LocalizationStore, LocalizationStoreError},
+    manager::{http_client::ManagerHTTPClientError, InstallationPhase, ManagerHTTPClient},
+    network::{NetworkStore, NetworkStoreError},
+    product::{ProductStore, ProductStoreError},
+    scripts::{ScriptsClient, ScriptsClientError, ScriptsGroup, ScriptsStore, ScriptsStoreError},
+    software::{SoftwareStore, SoftwareStoreError},
+    storage::{
+        http_client::{ISCSIHTTPClient, ISCSIHTTPClientError},
+        StorageStore, StorageStoreError,
+    },
+    users::{UsersStore, UsersStoreError},
 };
 
+#[derive(Debug, thiserror::Error)]
+pub enum StoreError {
+    #[error(transparent)]
```

Comment:
does it make sense to have it transparent here? I think additional line can help with seeing which part of store failed

Reply:
Each of the errors already contain a description (e.g., "Error processing storage settings: {0}").